CDTDatastore/touchdb/TDReplicator.m

```diff
@@ -819,9 +907,14 @@ - (void)saveLastSequence
                       // read-only and don't attempt to read its checkpoint next time.
                   } else if (_db) {
                       id rev = response[@"rev"];
+                      id ID = response[@"id"];
                       if (rev) body[@"_rev"] = rev;
+                      if (ID) body[@"_id"] = ID;
                       self.remoteCheckpoint = body;
-                      [_db setLastSequence:_lastSequence withCheckpointID:checkpointID];
+                      if (ID && [self.db saveCheckpointDocument:body error:nil]) {
+                          CDTLogWarn(CDTREPLICATION_LOG_CONTEXT,
```

Comment:
We have had issues before when we just log when things go wrong instead of propagating an error. I guess this is an awkward one because failing to checkpoint won't have impacted the completion of the replication, but will have implications for the next replication.
Do you think we should error here?